homeassistant/components/zwave/__init__.py

```diff
@@ -499,7 +501,19 @@ def rename_node(service):
         node.name = name
         _LOGGER.info(
             "Renamed Z-Wave node %d to %s", node_id, name)
+        update_ids = service.data.get(const.ATTR_UPDATE_IDS)
+        # We want to rename the device, the node entity,
+        # and all the contained entities
+        node_key = 'node-{}'.format(node_id)
+        entity = hass.data[DATA_DEVICES][node_key]
+        hass.async_create_task(entity.node_renamed(update_ids))
```

Comment:
Let's not create many tasks, instead, just make `rename_node` async instead of a callback so you can await it .